core-rust/mesh-api-server/src/mesh_api/handlers/block.rs

```diff
@@ -6,7 +6,7 @@ pub(crate) async fn handle_block(
 ) -> Result<Json<models::BlockResponse>, ResponseError> {
     assert_matching_network(&request.network_identifier, &state.network)?;
 
-    let database = state.state_manager.database.snapshot();
+    let database = state.state_manager.database.access_direct();
```

Comment:
I think direct access is okay here, but if we use direct access rather than snapshot, we should add a  comment explaining why.

e.g.
```
// All the block state is immutable once written, so we can use `access_direct` and save the overhead of a snapshot.
```

Reply:
Decided to revert to `snapshot()` as per [this conversation](https://rdxworks.slack.com/archives/C071C6DB8TG/p1733841098517859?thread_ts=1732877188.486819&cid=C071C6DB8TG)